src/Symfony/Component/HttpKernel/DataCollector/RequestDataCollector.php

```diff
@@ -435,14 +435,23 @@ private function parseController(array|object|string|null $controller): array|st
 
         if (\is_array($controller)) {
             try {
+                $rc = new \ReflectionClass($controller[0]);
                 $r = new \ReflectionMethod($controller[0], $controller[1]);
 
-                return [
+                $data = [
                     'class' => \is_object($controller[0]) ? get_debug_type($controller[0]) : $controller[0],
                     'method' => $controller[1],
                     'file' => $r->getFileName(),
                     'line' => $r->getStartLine(),
                 ];
+
+                if ($r->getFileName() !== $rc->getFileName()) {
+                    $data['abstractClass'] = $r->class;
```

Comment:
the name looks weird to me as this class might not be abstract. Maybe `methodClass`

Reply:
i agree. `methodClass` is much better.